nerfstudio/data/utils/nerfstudio_collate.py

```diff
@@ -163,11 +163,38 @@ def nerfstudio_collate(batch: Any, extra_mappings: Union[Dict[type, Callable], N
             op = torch.cat
 
         # Create metadata dictionary
-        metadata_keys = batch[0].metadata.keys()
-        assert all(
-            (cam.metadata.keys() == metadata_keys for cam in batch)
-        ), "All cameras must have the same metadata keys."
-        metadata = {key: op([cam.metadata[key] for cam in batch], dim=0) for key in metadata_keys}
+        if batch[0].metadata is not None:
+            metadata_keys = batch[0].metadata.keys()
+            assert all(
+                (cam.metadata.keys() == metadata_keys for cam in batch)
+            ), "All cameras must have the same metadata keys."
+            metadata = {key: op([cam.metadata[key] for cam in batch], dim=0) for key in metadata_keys}
+        else:
+            assert all((cam.metadata is None for cam in batch)), "All cameras must have the same metadata keys."
+            metadata = None
+
+        if batch[0].distortion_params is not None:
+            assert all(
```

Comment:
I believe this assertion for distortion_params is redundant since at the beginning of the function, we have 
```
        assert all((cam.distortion_params is None for cam in batch)) or all(
            (cam.distortion_params is not None for cam in batch)
```